src/main/java/seedu/address/logic/commands/CompareCalendarByTagCommand.java

```diff
@@ -0,0 +1,45 @@
+package seedu.address.logic.commands;
+
+import static java.util.Objects.requireNonNull;
+
+import java.util.List;
+
+import seedu.address.logic.commands.exceptions.CommandException;
+import seedu.address.model.Model;
+import seedu.address.model.calendar.ReadOnlyCalendar;
+import seedu.address.model.person.Person;
+import seedu.address.model.tag.Tag;
+
+/**
+ * Command to compare calendars of user with people in the calendar with some collection of tags.
+ */
+public class CompareCalendarByTagCommand extends Command {
+    public static final String COMMAND_WORD = "compareTag";
+    public static final String MESSAGE_USAGE = COMMAND_WORD + ": compare calendars within people with specified tags. "
+            + "Parameters: "
+            + "ONE/MULTIPLE VALID TAGS \n"
+            + "Example: " + COMMAND_WORD + " "
+            + "class friend";
+
+    public static final String MESSAGE_SUCCESS = "Displaying available time periods";
+
+    private final List<Tag> tagList;
+
+    public CompareCalendarByTagCommand(List<Tag> tagList) {
+        this.tagList = tagList;
```

Comment:
Might want to check what happens if the tagList is empty? What if there are no arguments? Or do a requireNonNull.